src/axolotl/prompt_strategies/completion.py

```diff
@@ -53,8 +53,8 @@ def tokenize_prompt(self, prompt):
             tokenized_full_prompt = self._tokenize(full_prompt)
 
             for key, val in tokenized_full_prompt.items():
-                for i in range(0, len(val), self.sequence_len):
-                    res[key].append(val[i : i + self.sequence_len])
+                for i in range(0, len(val), self.max_length):
+                    res[key].append(val[i : i + self.max_length])
```

Comment:
there is a difference in the completion prompts between the sequence len and the max length as they aren't the same in this case. the completion prompter tokenizes at a longer length and then splits the text into correct length

Reply:
Perhaps, we should also document this (as comments?) as it would be confusing to others as they read the code.

Reply:
Could you please provide more details? I'm still having trouble understanding. I will leave a comment here if needed.

Reply:
reverted the change that I unified sequence_len and max_length. PTAL

Reply:
@winglian do you still have a concern or can you review this PR?